Frontend/android-native/mobility-common/src/main/java/in/juspay/mobility/common/carousel/VPAdapter.java

```diff
@@ -0,0 +1,244 @@
+package in.juspay.mobility.common.carousel;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.res.Resources;
+import android.graphics.Color;
+import android.graphics.drawable.GradientDrawable;
+import android.text.Html;
+import android.util.Log;
+import android.view.Gravity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.webkit.JavascriptInterface;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.YouTubePlayer;
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.listeners.AbstractYouTubePlayerListener;
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.listeners.YouTubePlayerFullScreenListener;
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.listeners.YouTubePlayerListener;
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.options.IFramePlayerOptions;
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.views.YouTubePlayerView;
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.ui.DefaultPlayerUiController;
+
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+
+import in.juspay.hyper.core.ExecutorManager;
+import in.juspay.mobility.app.R;
+import in.juspay.mobility.common.YoutubeVideoView;
+
+public class VPAdapter extends RecyclerView.Adapter<VPAdapter.ViewHolder> {
+    ArrayList<ViewPagerItem> viewPagerItemArrayList;
+    public static float videoDuration = 0;
+
+    public static YouTubePlayerView youTubePlayerView ;
+    public static YouTubePlayer youtubePlayer;
+    public static Context context ;
+    public VPAdapter(ArrayList<ViewPagerItem> viewPagerItemArrayList, Context context) {
+        this.viewPagerItemArrayList = viewPagerItemArrayList;
+        this.context = context;
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.viewpager_item,parent,false);
+        return new ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+        try{
```

Comment:
Why try/ catch block? Try not to use try catch block frequently

Reply:
this is required.